packages/fx-core/src/core/FxCore.ts

```diff
@@ -1143,16 +1150,65 @@ export class FxCore {
     const context = createContextV3();
     return await coordinator.publishInDeveloperPortal(context, inputs as InputsWithProjectPath);
   }
+
+  async injectCreateAPIKeyAction(ymlPath: string, authName: string): Promise<void> {
+    const ymlContent = await fs.readFile(ymlPath, "utf-8");
+
+    const document = parseDocument(ymlContent);
+    const provisionNode = document.get("provision") as any;
+
+    if (provisionNode) {
+      const hasApiKeyAction = provisionNode.items.some(
+        (item: any) =>
+          item.get("uses") === "apiKey/create" && item.get("with")?.get("name") === authName
+      );
+
+      if (!hasApiKeyAction) {
+        provisionNode.items = provisionNode.items.filter(
+          (item: any) => item.get("uses") !== "apiKey/create"
+        );
+        let added = false;
+        for (let i = 0; i < provisionNode.items.length; i++) {
+          const item = provisionNode.items[i];
+          if (item.get("uses") === "teamsApp/create") {
+            const teamsAppId = item.get("writeToEnvironmentFile")?.get("teamsAppId") as string;
+            if (teamsAppId) {
+              provisionNode.items.splice(i + 1, 0, {
+                uses: "apiKey/create",
+                with: {
+                  name: `${authName}`,
+                  appId: `\${{${teamsAppId}}}`,
+                  apiSpecPath: "./apiSpecificationFiles/openapi.yaml",
```

Comment:
should we read the path of apiSpecFile from manifest.json?

Reply:
OK, thanks~